pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -170,6 +178,34 @@ public static PropertiesConfiguration fromFile(@Nullable File file,
     return config;
   }
 
+  /**
+   * Instantiate a {@link PropertiesConfiguration} from a {@link File}.
+   * @param file containing properties
+   * @param setDefaultDelimiter representing to set the default list delimiter.
+   * @param ioFactoryKind representing to set IOFactory. It can be null.
+   * @param failOnDuplicateKeys fail the loading when seeing duplicate keys.
+   * @return a {@link PropertiesConfiguration} instance.
+   */
+  public static PropertiesConfiguration fromFile(@Nullable File file, boolean setDefaultDelimiter,
+      @Nullable PropertyIOFactoryKind ioFactoryKind, boolean failOnDuplicateKeys)
```

Comment:
(minor) Can we move all handling into this method, and make the one on line 169 call this method with `failOnDeplicateKeys = false`?